ibis/backends/dask/execution/arrays.py

```diff
@@ -53,12 +38,60 @@ def execute_array_column(op, cols, **kwargs):
     )
 
 
+@execute_node.register(ops.ArrayConcat, dd.Series, dd.Series)
+def execute_array_concat_series(op, left, right, **kwargs):
+    df = dd.concat([left, right], axis=1)
+    return df.apply(
+        lambda row: np.concatenate(row), axis=1, meta=dd.utils.make_meta(left),
+    )
+
+
+@execute_node.register(ops.ArrayConcat, dd.Series, np.ndarray)
+@execute_node.register(ops.ArrayConcat, np.ndarray, dd.Series)
+def execute_array_concat_mixed(op, left, right, **kwargs):
+    if isinstance(left, dd.Series):
+        # "Iterate" over `left` Series, concatenating each array/element
+        # in the Series with `right`, which is a single ndarray.
+        return left.apply(
+            lambda arr: np.concatenate([arr, right]),
+            meta=dd.utils.make_meta(left),
+        )
+    elif isinstance(right, dd.Series):
+        return right.apply(
+            lambda arr: np.concatenate([left, arr]),
+            meta=dd.utils.make_meta(right),
+        )
+
+
+@execute_node.register(ops.ArrayConcat, np.ndarray, np.ndarray)
+def execute_array_concat_scalar(op, left, right, **kwargs):
+    return np.concatenate([left, right])
+
+
+@execute_node.register(ops.ArrayRepeat, dd.Series, int)
+def execute_array_repeat(op, data, n, **kwargs):
+    # Negative n will be treated as 0 (repeat will produce empty array)
+    n = max(n, 0)
+    return data.apply(
+        lambda arr: np.repeat(arr, n), meta=dd.utils.make_meta(data)
+    )
+
+
+@execute_node.register(ops.ArrayRepeat, np.ndarray, int)
+def execute_array_repeat_scalar(op, data, n, **kwargs):
+    # Negative n will be treated as 0 (repeat will produce empty array)
+    return np.repeat(data, max(n, 0))
+
+
 # TODO - aggregations - #2553
+# Currently, Summarize aggcontext with non-string `function` does not
+# work on Dask backend, because it tries to call `dd.Series.agg`, which
+# does not exist.
 @execute_node.register(ops.ArrayCollect, dd.Series)
 def execute_array_collect(op, data, aggcontext=None, **kwargs):
-    return aggcontext.agg(data, collect_list)
+    return aggcontext.agg(data, np.array)
```

Comment:
Does using `data.agg(collect_array)` not work here?

Reply:
Ah this does not because `dd.Series` doesn't have `.agg` implemented (same issue that I mentioned before with the `Summarize` aggcontext)

Reply:
Ah - I think we just want to do https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.Series.to_dask_array here (`dd.Series -> dd.array`). Though this is another place where I think the dask backend needs to work with `dd.array` rather than `np.array`.

Reply:
Yeah - I don't know what we could do here besides triggering computation. Depending on what Jeff/Li think, I think that might be the right thing to do. Otherwise I'm not sure how this could work 

```
t.mutate(
    my_arr_col=my_int_col.collect()
).execute()
```